lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -257,18 +255,10 @@ static ConstructorDecl *deriveRawRepresentable_init(TypeChecker &tc,
   auto rawType = ArchetypeBuilder::mapTypeIntoContext(parentDC,
                                                       rawInterfaceType);
 
-  // Make sure that the raw type is Equatable. We need it to ensure that we have
-  // a suitable ~= for the switch.
   auto equatableProto = tc.getProtocol(enumDecl->getLoc(),
                                        KnownProtocolKind::Equatable);
-  if (!equatableProto)
-    return nullptr;
-
-  if (!tc.conformsToProtocol(rawType, equatableProto, enumDecl, None)) {
-    SourceLoc loc = enumDecl->getInherited()[0].getSourceRange().Start;
-    tc.diagnose(loc, diag::enum_raw_type_not_equatable, rawType);
-    return nullptr;
-  }
+  assert(equatableProto);
```

Comment:
This will emit an unused variable warning if asserts are off.

Add a `(void) equatableProto;` right after to silence this (and confirm by compiling with `./build-script -r`)